src/drivers/file/server.rs

```diff
@@ -0,0 +1,98 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use chrono::DateTime;
+use mavlink::ardupilotmega::MavMessage;
+use std::path::PathBuf;
+use tokio::io::AsyncReadExt;
+use tokio::sync::broadcast;
+use tracing::*;
+
+use crate::drivers::{Driver, DriverInfo};
+use crate::protocol::Protocol;
+
+#[derive(Clone, Debug)]
+pub struct FileServer {
+    pub path: PathBuf,
+}
+
+impl FileServer {
+    #[instrument(level = "debug")]
+    pub fn try_new(file_path: &str) -> Result<Self> {
+        let path = PathBuf::from(file_path);
+        Ok(Self { path })
+    }
+
+    #[instrument(level = "debug", skip(reader, hub_sender))]
+    async fn handle_client(
+        server: FileServer,
+        mut reader: tokio::io::BufReader<tokio::fs::File>,
+        hub_sender: Arc<broadcast::Sender<Protocol>>,
+    ) -> Result<()> {
+        let source_name = server.path.as_path().display().to_string();
+        loop {
+            let Ok(us_since_epoch) = reader.read_u64().await else {
+                info!("End of file reached");
+                break;
+            };
+
+            let Some(_date_time) = DateTime::from_timestamp_micros(us_since_epoch as i64) else {
+                warn!("Failed to convert unix time");
+                continue;
+            };
+
+            if (reader.buffer().is_empty()) {
+                info!("End of file reached");
+                break;
+            }
+
+            if (reader.buffer()[0] != mavlink::MAV_STX_V2) {
+                warn!("Invalid MAVLink start byte, skipping");
+                continue;
+            }
```

Comment:
we can ditch this: the read_v2_raw_message_async already handles this

Reply:
Check comment.